pkg/log/api_test.go

```diff
@@ -0,0 +1,228 @@
+package log
+
+import (
+	"context"
+	"testing"
+
+	"github.com/anz-bank/sysl/pkg/log/loggers"
+	"github.com/anz-bank/sysl/pkg/log/testutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestWithLogger(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	logger := loggers.NewMockLogger()
+	logger.On("Copy").Return(loggers.NewMockLogger()).Once()
+	newCtx := WithLogger(ctx, logger)
+	newLogger := newCtx.Value(loggerKey).(*loggers.MockLogger)
+
+	require.True(t, logger.AssertExpectations(t))
+	assert.True(t, &ctx != &newCtx)
+	assert.True(t, &logger != &newLogger)
+}
+
+func TestWithField(t *testing.T) {
+	testSingleFieldCases(t, "TestWithField",
+		func(tt *testing.T, ctx context.Context, sc testutil.SingleField) {
+			logger := ctx.Value(loggerKey).(*loggers.MockLogger)
+			newCtx := WithField(ctx, sc.Key, sc.Val)
+			newLogger := newCtx.Value(loggerKey).(*loggers.MockLogger)
+
+			require.True(tt, logger.AssertExpectations(tt))
+			assert.True(tt, &ctx != &newCtx)
+			assert.True(tt, &logger != &newLogger)
```

Comment:
You are comparing the addresses of variables, which will never be equal.

Reply:
after looking through the tests again, I realized that those are values of returned from a mock interface. It might be pointless to test that, will remove these tests.